modin/pandas/base.py

```diff
@@ -1396,8 +1397,15 @@ def drop_duplicates(
             duplicates = self.duplicated(keep=keep, subset=subset)
         else:
             duplicates = self.duplicated(keep=keep)
-        indices = duplicates.values.nonzero()[0]
-        return self.drop(index=self.index[indices], inplace=inplace)
+
+        result = self[-duplicates]
+        if ignore_index:
+            result.index = list(range(len(result)))
```

Comment:
`list(range(len(results)))` doesn't feel very elegant.
Maybe `pandas.RangeIndex(stop=len(result))` instead? This may better match what Pandas does.

Reply:
yeah, I decided I liked that better.